src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -63,6 +63,12 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
             // We would like to be sure that nobody changes buffers at the same time.
             AssertIsForeground();
 
+            if (_textView.Selection.Mode == TextSelectionMode.Box)
```

Comment:
💭 It seems like this should be a separate issue. Do we have tests for this?

Reply:
@sharwell, the second fix was a caused by a test failure. This test:
https://github.com/dotnet/roslyn/blob/43368ee4df7f26b693c68d339500f3f1e15427bf/src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb#L2925
worked occasionally due to missing of those two checks.
When I restored one of conditions, it started failing. It requires to restore
`if (_textView.Selection.Mode == TextSelectionMode.Box)`
from the old completion.